src/parser/parsers/heroes.py

```diff
@@ -104,11 +104,60 @@ def _parse_hero_weapon(self, hero_value):
             'RoundsPerSecond': 1 / weapon_prim['m_flCycleTime'],
             'ClipSize': weapon_prim['m_iClipSize'],
             'ReloadTime': weapon_prim['m_reloadDuration'],
+            'ReloadMovespeed': int(weapon_prim['m_flReloadMoveSpeed']) / 10000,
+            'ReloadDelay': weapon_prim['m_flReloadSingleBulletsInitialDelay'] if 'm_flReloadSingleBulletsInitialDelay' in weapon_prim else 0,
+            'ReloadSingle': weapon_prim['m_bReloadSingleBullets'] if 'm_bReloadSingleBullets' in weapon_prim else False,
+            'BulletSpeed': self.calc_bullet_velocity(weapon_prim['m_BulletSpeedCurve']['m_spline']),
+            'FalloffStartRange': weapon_prim['m_flDamageFalloffStartRange'] / ENGINE_UNITS_PER_METER,
+            'FalloffEndRange': weapon_prim['m_flDamageFalloffEndRange'] / ENGINE_UNITS_PER_METER,
+            'FalloffStartScale': weapon_prim['m_flDamageFalloffStartScale'],
+            'FalloffEndScale': weapon_prim['m_flDamageFalloffEndScale'],
+            'FalloffBias': weapon_prim['m_flDamageFalloffBias'],
+            'BulletGravityScale': weapon_prim['m_flBulletGravityScale'],
+            'BulletRadius': weapon_prim['m_flBulletRadius'] / ENGINE_UNITS_PER_METER,
         }
 
         weapon_stats['DPS'] = weapon_stats['BulletDamage'] * weapon_stats['RoundsPerSecond']
         return weapon_stats
 
+    def calc_bullet_velocity(self, spline):
+        """Calculates bullet velocity of a spline, ensuring its linear"""
+        """
+        Transforms
+        [
+            {
+                "x": 0.0,
+                "y": 23999.998047,
+                "m_flSlopeIncoming": 0.0,
+                "m_flSlopeOutgoing": 0.0
+            },
+            {
+                "x": 100.0,
+                "y": 23999.998047,
+                "m_flSlopeIncoming": 0.0,
+                "m_flSlopeOutgoing": 0.0
+            }
+        ]
+
+        to
+
+        23999.998047
+        """
+
+        # Confirm its linear
+        for point in spline:
+            if point['m_flSlopeIncoming'] != 0 or point['m_flSlopeOutgoing'] != 0:
+                raise Exception('Bullet speed curve is not linear')
+
+        # Confirm its constant
+        last_y = spline[0]['y']
+        for point in spline:
+            if point['y'] != last_y:
+                raise Exception('Bullet speed curve is not constant')
+
+        # If constant, return the y
+        return last_y / ENGINE_UNITS_PER_METER
```

Comment:
I think we should round the bullet speed, as its been calculated empirically. So the many decimal places is a little misleading as to what our precision is